scipy-stubs/signal/_peak_finding.pyi

```diff
@@ -1,37 +1,72 @@
-from scipy._typing import Untyped
+from collections.abc import Callable, Sequence
+from typing import Concatenate, Literal, TypeAlias, TypedDict
+from typing_extensions import NotRequired
+
+import numpy as np
+import numpy.typing as npt
 
 __all__ = ["argrelextrema", "argrelmax", "argrelmin", "find_peaks", "find_peaks_cwt", "peak_prominences", "peak_widths"]
 
-def argrelmin(data: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def argrelmax(data: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def argrelextrema(data: Untyped, comparator: Untyped, axis: int = 0, order: int = 1, mode: str = "clip") -> Untyped: ...
-def peak_prominences(x: Untyped, peaks: Untyped, wlen: Untyped | None = None) -> Untyped: ...
+_Mode: TypeAlias = Literal["clip", "wrap"]
+_ProminencesResult: TypeAlias = tuple[npt.NDArray[np.float64], npt.NDArray[np.intp], npt.NDArray[np.intp]]
+_WidthsResult: TypeAlias = tuple[
+    npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64], npt.NDArray[np.float64]
+]
```

Comment:
I've seen quite a lot of `npt.NDArray[np.int64]` and `npt.NDArray[np.inpt]` being used in this module, so perhaps it could help to introduce type aliases for these two as well